lib/msf/ui/debug.rb

```diff
@@ -145,7 +148,56 @@ def self.datastore(framework, driver)
           content
         )
       rescue StandardError => e
-        section_build_error('Failed to extract Datastore', e)
+        build_section(
+          'Module/Datastore',
+          ERROR_BLURB,
+          section_build_error('Failed to extract Datastore', e)
+        )
+      end
+
+      def self.database_configuration(framework)
+        output = "```\nSession Type: #{db_connection_info(framework)}\n```\n\n"
+
+        if framework.db&.active
+          output += "| ID | Hosts | Vulnerabilities | Notes | Services |\n"
+          output += "|-:|-:|-:|-:|-:|\n"
+
```

Comment:
Maybe something like:

```ruby
current_workspace = framework.db.workspace
example_workspaces = ::Mdm::Workspace.order(id: :desc).take(10)
ordered_workspaces = ([current_workspace] + example_workspaces).uniq.sort_by(&:id)
workspace_rows = ordered_workspaces.map do |workspace|
  id = current_workspace.id == workspace.id ? "#{workspace.id} **(Current)**" : workspace.id
  [
    id,
    workspace.hosts.count.to_s(:delimited),
    workspace.vulns.count.to_s(:delimited),
    workspace.notes.count.to_s(:delimited),
    workspace.services.count.to_s(:delimited)
  ]
end

totals_row = [
  "**Total (#{::Mdm::Workspace.count.to_s(:delimited)})**",
  "**#{::Mdm::Host.count.to_s(:delimited)}**",
  "**#{::Mdm::Vuln.count.to_s(:delimited)}**",
  "**#{::Mdm::Note.count.to_s(:delimited)}**",
  "**#{::Mdm::Service.count.to_s(:delimited)}**"
]

table = "| ID | Hosts | Vulnerabilities | Notes | Services |\n"
table += "|-:|-:|-:|-:|-:|\n"
table += (workspace_rows + [totals_row]).map { |x| "| #{x.join(" | ")} |" }.join("\n")
```

So for me that would look like this:

| ID | Hosts | Vulnerabilities | Notes | Services |
|-:|-:|-:|-:|-:|
| 7 | 0 | 0 | 0 | 0 |
| 8 | 0 | 0 | 0 | 0 |
| 9 | 0 | 0 | 0 | 0 |
| 10 | 0 | 0 | 0 | 0 |
| 11 | 0 | 0 | 0 | 0 |
| 12 | 0 | 0 | 0 | 0 |
| 13 | 0 | 0 | 0 | 0 |
| 14 **(Current)** | 1 | 288 | 2 | 21 |
| 15 | 0 | 0 | 0 | 0 |
| 16 | 0 | 0 | 0 | 0 |
| **Total (16)** | **9** | **867** | **18** | **69** |

Whilst before it looked like this:

| ID | Hosts | Vulnerabilities | Notes | Services |
|-:|-:|-:|-:|-:|
| 14 **(Current)** | 1 | 288 | 2 | 21 |
| 1 | 7 | 291 | 14 | 27 |
| 2 | 1 | 288 | 2 | 21 |
| 3 | 0 | 0 | 0 | 0 |
| 4 | 0 | 0 | 0 | 0 |
| 5 | 0 | 0 | 0 | 0 |
| 6 | 0 | 0 | 0 | 0 |
| 7 | 0 | 0 | 0 | 0 |
| 8 | 0 | 0 | 0 | 0 |
| 9 | 0 | 0 | 0 | 0 |
| **Total (16)** | **9** | **867** | **18** | **69** |

I sorted the rows to ensure we can see examples of the most recent workspaces, rather than the first 10. Although I could be convinced ascending order is better 🤔

Reply:
Thanks for the feedback.
I agree, your approach is a lot better so I will change it over 👍 

What do you think about also including the default workspace? 👀  Would it be relevant/useful? (I might need to do further research on this to see how default workspaces are handled, but I think it _could_ be useful?)